apps/wallet/src/ui/app/pages/home/transfer-coin/TransferCoinForm/StepOne.tsx

```diff
@@ -1,111 +1,158 @@
-import { useEffect, useRef, memo } from 'react';
+import { useEffect, useRef } from 'react';
 
-import { Content, Menu } from '_app/shared/bottom-menu-layout';
-import Button from '_app/shared/button';
-import ActiveCoinsCard from '_components/active-coins-card';
+import { Text } from '_app/shared/text';
+import { IconTooltip } from '_app/shared/tooltip';
+import AddressInput from '_components/address-input';
 import Alert from '_components/alert';
-import Icon, { SuiIcons } from '_components/icon';
 import NumberInput from '_components/number-input';
 import { parseAmount } from '_helpers';
-import { useCoinDecimals } from '_hooks';
+import { useCoinDecimals, useFormatCoin } from '_hooks';
+import { GAS_SYMBOL, GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
 
 import type { FormValues } from '../';
 
-import st from './TransferCoinForm.module.scss';
-
 export type TransferCoinFormProps = {
-    coinSymbol: string;
+    submitError: string | null;
     coinType: string;
+    gasCostEstimation: number | null;
+    gasEstimationLoading?: boolean;
     onClearSubmitError: () => void;
     onAmountChanged: (amount: bigint) => void;
+    balance: bigint | null;
 };
 
-function StepOne({
-    coinSymbol,
+export function StepOne({
+    submitError,
     coinType,
     onClearSubmitError,
     onAmountChanged,
+    gasCostEstimation,
+    gasEstimationLoading,
+    balance,
 }: TransferCoinFormProps) {
     const {
-        isValid,
         validateForm,
-        values: { amount },
+        values: { amount, to },
         errors,
         touched,
+        setFieldValue,
     } = useFormikContext<FormValues>();
     const onClearRef = useRef(onClearSubmitError);
     onClearRef.current = onClearSubmitError;
     useEffect(() => {
         onClearRef.current();
-    }, [amount]);
+    }, [amount, to]);
     const [coinDecimals, { isLoading: isCoinDecimalsLoading }] =
         useCoinDecimals(coinType);
+
     useEffect(() => {
         if (!isCoinDecimalsLoading) {
             const parsedAmount = parseAmount(amount, coinDecimals);
             onAmountChanged(parsedAmount);
-            // seems changing the validationSchema doesn't rerun the validation for the form
-            // trigger re-validation here when the amount to send is changed
-            // (changing the amount will probably change the gasBudget and in the end the validationSchema)
-            validateForm();
         }
-    }, [
-        amount,
-        onAmountChanged,
-        coinDecimals,
-        isCoinDecimalsLoading,
-        validateForm,
-    ]);
+    }, [amount, coinDecimals, isCoinDecimalsLoading, onAmountChanged]);
+
+    const [formattedGas] = useFormatCoin(gasCostEstimation, GAS_TYPE_ARG);
+    const [maxToken, symbol, queryResult] = useFormatCoin(balance, coinType);
+
+    // set sendMaxToken to true if the amount is equal to the maxToken
+    useEffect(() => {
+        setFieldValue('sendMaxToken', amount === maxToken);
```

Comment:
I am confused. You mean the max is not the max if the user manually enter the exact amount of coins available?

Reply:
we discussed about it in a call - My point is that we should only change it to sendMax/paySuiAll only when user clicks Max and not by inserting the max value in the input. When the user changes the input value it will always be an exact amount

Reply:
I made the change so that it just uncheck the max button once the input is updated